<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form0</class>
 <widget class="QMainWindow" name="Form0">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>617</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Компьютерная демонстрация - Радуга</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(0, 0, 0);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QLabel" name="header">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="font">
       <font>
        <pointsize>25</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 255, 255);</string>
      </property>
      <property name="text">
       <string>Ход лучей в капле</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="text">
             <string>Прицельный параметр</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="spin1">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="minimum">
             <double>-0.990000000000000</double>
            </property>
            <property name="maximum">
             <double>0.990000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.800000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSlider" name="slider1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="minimum">
           <number>-99</number>
          </property>
          <property name="value">
           <number>80</number>
          </property>
          <property name="sliderPosition">
           <number>80</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label2">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="text">
             <string>Длина волны</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spin2">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="minimum">
             <number>400</number>
            </property>
            <property name="maximum">
             <number>760</number>
            </property>
            <property name="value">
             <number>600</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSlider" name="slider2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="minimum">
           <number>400</number>
          </property>
          <property name="maximum">
           <number>760</number>
          </property>
          <property name="value">
           <number>600</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addbeam">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>60</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(170, 170, 170);</string>
          </property>
          <property name="text">
           <string>&amp;Добавить луч</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="prev">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>60</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>12</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(170,170,170);</string>
            </property>
            <property name="text">
             <string>&lt; Ход луча</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="next">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>60</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>12</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(170,170,170);</string>
            </property>
            <property name="text">
             <string>Ход луча &gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(255, 255, 255);</string>
          </property>
          <property name="text">
           <string>Показать все лучи</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(255, 255, 255);</string>
          </property>
          <property name="text">
           <string>Первичная радуга</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_2">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(255, 255, 255);</string>
          </property>
          <property name="text">
           <string>Вторичная радуга</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="clearbut">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>60</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(170,170,170);</string>
          </property>
          <property name="text">
           <string>&amp;Очистить экран</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="quit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(170,170,170);</string>
          </property>
          <property name="text">
           <string>&amp;Выход</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="GLWidget" name="widget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>quit</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addbeam</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>add_beam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>next</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>incN()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>171</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>244</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>prev</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>decN()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>radio1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>326</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>radio3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>175</x>
     <y>326</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>radio2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider2</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Form0</receiver>
   <slot>wave_slider(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider1</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Form0</receiver>
   <slot>dist_slider(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>54</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>253</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form0</receiver>
   <slot>wave_spin(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin1</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Form0</receiver>
   <slot>dist_spin(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearbut</sender>
   <signal>clicked()</signal>
   <receiver>Form0</receiver>
   <slot>clearscr()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>397</y>
    </hint>
    <hint type="destinationlabel">
     <x>87</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>add_beam()</slot>
  <slot>radio1()</slot>
  <slot>radio2()</slot>
  <slot>radio3()</slot>
  <slot>incN()</slot>
  <slot>decN()</slot>
  <slot>dist_slider(int)</slot>
  <slot>wave_slider(int)</slot>
  <slot>dist_spin(double)</slot>
  <slot>wave_spin(int)</slot>
  <slot>clearscr()</slot>
 </slots>
</ui>
